src/plugins/visualizations/server/usage_collector/get_usage_collector.ts

```diff
@@ -35,61 +34,52 @@ export interface VisualizationUsage {
  * Parse the response data into telemetry payload
  */
 export async function getStats(
-  esClient: ElasticsearchClient,
-  index: string
+  soClient: SavedObjectsClientContract
 ): Promise<VisualizationUsage | undefined> {
-  const searchParams = {
-    size: 10000, // elasticsearch index.max_result_window default value
-    index,
-    ignoreUnavailable: true,
-    filterPath: [
-      'hits.hits._id',
-      'hits.hits._source.visualization',
-      'hits.hits._source.updated_at',
-    ],
-    body: {
-      query: {
-        bool: { filter: { term: { type: 'visualization' } } },
-      },
-    },
-  };
-  const { body: esResponse } = await esClient.search<ESResponse>(searchParams);
-  const size = get(esResponse, 'hits.hits.length', 0);
-  if (size < 1) {
-    return;
-  }
-
-  // `map` to get the raw types
-  const visSummaries: VisSummary[] = esResponse.hits.hits.map((hit) => {
-    const spacePhrases = hit._id.toString().split(':');
-    const lastUpdated: string = get(hit, '_source.updated_at');
-    const space = spacePhrases.length === 3 ? spacePhrases[0] : 'default'; // if in a custom space, the format of a saved object ID is space:type:id
-    const visualization = get(hit, '_source.visualization', { visState: '{}' });
-    const visState: { type?: string } = JSON.parse(visualization.visState);
-    return {
-      type: visState.type || '_na_',
-      space,
-      past_days: getPastDays(lastUpdated),
-    };
+  const finder = await soClient.createPointInTimeFinder({
```

Comment:
my question applies to all refactored telemetries (#98903, #98914, #99023) where we replaced `esClient` with `soClient`. 
During my testing today, I found that this `finder` returns only data for the `default` namespace. If I create visualizations for a `non-default` namespace, this will not be reflected in the telemetry.  

I've tried to fix that by adding `namespaces: ['*'],` but with that argument pagination doesn't work. see:
![image](https://user-images.githubusercontent.com/20072247/116874946-cb068600-ac22-11eb-99e9-36bbfaa1cb28.png)

@lukeelmers @afharo could you please help me a little cause we have a recommendation to use `soClient` ( https://github.com/elastic/kibana/pull/88604#discussion_r560801214 )  for telemetry with pagination.

Reply:
Ah, yeah we specifically [decided against adding namespace support](https://github.com/elastic/kibana/pull/89915/files#r573754765) when using PIT until we had a concrete need for it. I guess we do now 🙂  

I've created an issue to track this enhancement: https://github.com/elastic/kibana/issues/99044

@afharo WDYT, do we push forward on this work without `soClient` (or with using traditional pagination which typically works with <10k results), or do we wait until such a time as we have namespace support in https://github.com/elastic/kibana/issues/99044?

Reply:
I vote to wait, because the code with `soClient` looks easier to me + we don't need to implement paging ourselves. If I'm not mistaken in `esClient` we don't have a simple wrapper like `createPointInTimeFinder` . + https://github.com/elastic/kibana/pull/88604#discussion_r560801214

Reply:
> do we push forward on this work without `soClient`

IMO, it feels like we keep coming back and forth with this conversation. And, usually, AFAIK, the general recommendation is to use the SO Client whenever possible because using _plain_ ES client requests would break if/when we change the internal implementations of SOs (underlying indices, ways to store them, ...).

I'm not fully familiar with the limitations of allowing `namespaces: ['*']`. I need to better understand the reasons exposed in #99044 before recommending migrating to `esClient` as a workaround :)

So ++ to wait :)

Reply:
I've been testing locally, and I think these changes would solve the issue: https://github.com/alexwizp/kibana/pull/6

I tested it locally and it works as expected to me :)

Reply:
@afharo I've tested your changes and not sure that it works. 

How to test: 
1) Start Kibana with `Trial` license 
2) Install some of out defaults data sets. 
3) Execute `/api/stats?extended=true&legacy=true` to get the actual status of telemetry
For my local I got: 
![image](https://user-images.githubusercontent.com/20072247/117135934-53fef800-adb0-11eb-9c5b-91f43f9857b8.png)
4) Go to `Stack Management / Spaces` and create a new space
5) Re-login into created space and create new `TSVB` visualization 
6) Execute `/api/stats?extended=true&legacy=true` again to get the actual status of telemetry

**Expected result:** expect to see **+1** visualization. 
**Actual result:** the result is the same as in **step 3**

Reply:
Also looks like we use soClient without TelemetrySavedObjectsClient wrapper. It's a reason why you don't see an error in console

Reply:
Hmm, interesting... This is how I tested it:

1. Start Kibana with `Trial` license
2. Install some of the defaults datasets
3. Go to `Stack Management / Advanced Settings`. 
4. At the bottom of the page, click on `See examples of cluster data`.
5. Find the output `visualization_types`.
6. Create a new Space and load the same default datasets
7. Repeat 3-5 and compare the results: they should double.

Reply:
> I'm not fully familiar with the limitations of allowing `namespaces: ['*']`.

I'm not super familiar with the telemetry SO client, but if this works for you, it sounds like it is circumventing the secure SO client wrapper in the security plugin, which should be explicitly forbidding this behavior (hence the error @alexwizp is seeing): https://github.com/elastic/kibana/blob/12b245c4e50555c227d569bc54bc64118bd15558/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts#L226-L230

When using `pit`, we don't have any testing for the multi-namespace scenario yet (this is something that would be done as a part of #99044).

> I need to better understand the reasons exposed in #99044 before recommending migrating to esClient as a workaround :)

To be clear, moving forward with `soClient` is still possible without using a `pit`, it just means we'd have to set a page size, and would be limited by the ES `index.max_result_window` (default is 10k). Which of course means this PR wouldn't really be resolving https://github.com/elastic/kibana/issues/93770, but rather simply doing the conversion to `soClient`.

Reply:
Nevermind, I don't know how I thought it worked when I tried... It's def not working 🤷 

Sorry about that! Then, we need to rely on the outcome of #99044

Reply:
So I guess the question remains, do we adjust the scope of this PR as follows?

> - ~~Size parameter was changed from 10000 to 1000~~
> - Code was refactored to use soClient instead of esClient
> - ~~The paging is used through the soClient.createPointInTimeFinder method~~
> - Code was optimized, extra cycles were removed

Basically, since @alexwizp has already started the work, is it worth merging with the same 10k size, only switching to SO client instead? WDYT @alexwizp @afharo

Reply:
I don't see a lot of sense just to switch `esClient` -> `soClient`.  We started that work to resolve #93770. 
But on the other hand I'm ok to use `soClient` with `size === 10000`. it adds more contributions to my piggy bank =)

Reply:
++ to what @alexwizp said. I guess the question is how long would it take us to fix #99044? If it's just a couple of weeks, probably it's worth putting this PR on hold and rebase when ready?

Reply:
I will add #99044 to our list of items to discuss for our next sprint. I don't think it would be a particularly large amount of work, it's mostly about deciding on a strategy with the security team & ensuring there's adequate test coverage.